druid/Dockerfile

```diff
@@ -2,95 +2,111 @@
 
 # Ignoring DL3038 globally because set `assumeyes=True` in dnf.conf in our base image
 # Ignoring DL4006 globally because we inherit the SHELL from our base image
-# hadolint global ignore=DL3038,DL4006
+# Ignoring SC2164 globally because we inherit SHELL from the base image which contains "-e" for bash
+# hadolint global ignore=DL3038,DL4006,SC2164
 
 # hadolint ignore=DL3006
-FROM stackable/image/java-devel as druid-builder
+FROM stackable/image/java-devel AS druid-builder
 
 ARG PRODUCT
 ARG JACKSON_DATAFORMAT_XML
 ARG STAX2_API
 ARG WOODSTOX_CORE
 ARG AUTHORIZER
 
-RUN microdnf update && \
-    microdnf install \
-    # Required to install pyyaml
-    python-pip \
-    # Required to patch druid
-    patch && \
-    microdnf clean all && \
-    rm -rf /var/cache/yum && \
-    # pyyaml is required for the compile Druid
-    pip install --no-cache-dir pyyaml==6.0.1
+# Setting this to anything other than "true" will keep the cache folders around (e.g. for Maven, NPM etc.)
+# This can be used to speed up builds when disk space is of no concern.
+ARG DELETE_CACHES="true"
```

Comment:
This can be moved to `druid/versions.py` now so it can be overridden with
```
bake --build-arg 'delete_caches=false' ...
```

Reply:
Oh...it needs to be in `versions.py` so the override works?
That's a bit annoying, I don't want to pollute `versions.py`

Reply:
Razvan and I talked and decided that this is a larger discussion that we don't have to have now.
I'll resolve this for now. It is usable for those that know how to...we can improve it later.